### PR TITLE
feat(validations): modify commitment in CommitTransaction to simplify…

### LINKED ISSUE #667: Validate Reveal Commitment

### PR DESCRIPTION
**Pull request by lrubiorod**
… validation

Modified commitment field in CommitTransaction from Hash to Signature
Modified protobuf schema of CommitTransaction
Modified mining data request process and validation for Reveal and Commit transactions
Deleted useless methods to create Commit, Reveal and Tally transactions
Created a function to provide the pkh from a PublicKey

Close #667

### CODE REVIEW

File: data_structures/src/transaction.rs [hunk 1]
```diff
@@ -183,7 +183,7 @@ pub struct CommitTransactionBody {
     //pub collateral: Vec<Input>, // ValueTransferOutput
     pub dr_pointer: Hash, // DTTransaction hash
     // Outputs
-    pub commitment: Hash,
+    pub commitment: Signature, // RevealTransaction Signature
```
Comment (anler) [inline data_structures/src/transaction.rs]: Better put the `// Reveal..` comment as documentation:

```
/// RevealTransaction Signature
 pub commitment: Signature,
```

File: validations/src/validations.rs [hunk 2]
```diff
@@ -264,13 +264,26 @@ pub fn validate_reveal_transaction(
...
     let dr_pointer = re_tx.body.dr_pointer;
-    let dr_output = dr_pool
-        .get_dr_output(&dr_pointer)
+    let dr_state = dr_pool
+        .data_request_pool
+        .get(&dr_pointer)
         .ok_or(TransactionError::DataRequestNotFound { hash: dr_pointer })?;
 
-    // TODO: Validate commitment
+    let pkh = re_tx.signatures[0].public_key.pkh();
```
Comment (anler) [inline validations/src/validations.rs]: You could make `validate_commit_reveal_signature` function to return a reference to the first signature and you would avoid the unsafe `signatures[0]` access here.

File: validations/src/validations.rs [hunk 3]
```diff
@@ -264,13 +264,26 @@ pub fn validate_reveal_transaction(
...
+
+    if let Some(commit) = dr_state.info.commits.get(&pkh) {
+        if commit.body.commitment != re_tx.signatures[0].signature {
+            Err(TransactionError::MismatchedCommitment)?
+        }
+    } else {
+        Err(TransactionError::CommitNotFound)?
+    }
```
Comment (anler) [inline validations/src/validations.rs]: You can un-nest things a little bit by using Option's `ok_or_else` method:
```rust
let commit = dr_state.info.commits.get(&pkh).ok_or_else(|| Err(TransactionError::CommitNotFound))?;
...
```

File: node/src/actors/chain_manager/mining.rs [hunk 4]
```diff
@@ -285,24 +284,27 @@ impl ChainManager {
...
-                        .map_err(|e| log::error!("Couldn't sign commit body: {}", e))
-                        .and_then(move |commit_signatures| {
-                            let reveal_body = create_reveal_body(dr_pointer, reveal_value, own_pkh);
-
-                            sign_transaction(&reveal_body, 1)
-                                .map(|reveal_signatures| {
+
+                    let reveal_body = RevealTransactionBody::new(dr_pointer, reveal_value, own_pkh);
+
+                    sign_transaction(&reveal_body, 1)
```
Comment (tmpolaczyk) [inline node/src/actors/chain_manager/mining.rs]: What do you think about adding a `create_commit_and_reveal` function to `signature_mngr`?

File: node/src/actors/chain_manager/mining.rs [hunk 5]
```diff
@@ -285,24 +284,27 @@ impl ChainManager {
...
+                        .map_err(|e| log::error!("Couldn't sign reveal body: {}", e))
+                        .and_then(move |reveal_signatures| {
+                            // Commitment is the RevealTransaction signature that will be published later
```
Comment (tmpolaczyk) [inline node/src/actors/chain_manager/mining.rs]:
```suggestion
                            // Commitment is the hash of the RevealTransaction signature that will be published later
```

File: data_structures/src/transaction.rs [hunk 6]
```diff
@@ -182,7 +182,7 @@ pub struct CommitTransactionBody {
     // TODO: Discussion about collateral
     //pub collateral: Vec<Input>, // ValueTransferOutput
     pub dr_pointer: Hash, // DTTransaction hash
```
Comment (tmpolaczyk) [inline data_structures/src/transaction.rs]: This should also be a doc comment
```suggestion
    /// DRTransaction hash
    pub dr_pointer: Hash,
```

**Pull request merged**